buildpacks/python/python/function.go

```diff
@@ -27,12 +27,12 @@ type Function struct {
 	funcYamlEnvs map[string]string
 }
 
-func NewFunction(opts ...FunctionOpt) Function {
-	f := Function{}
+func NewFunction(opts ...FunctionOpt) *Function {
```

Comment:
Why'd we switch to using pointers for the layers? I feel like you might have answered it in another comment but I couldn't find it.

Reply:
Here's the original thread. In summary, there are a few reasons:

- Convention: The term "new" sort of implies pointers (vs "make") -- details/links in that thread
- Consistency: Some of our layer contributors were already doing this, and one of them even had pointer receivers but the constructor was not returning a pointer

I don't feel too strongly which side we land on in the consistency bullet, as long as it's consistent. I tend to lean toward pointer receivers in this case because the contributors in general are somewhat stateful.

Reply:
But the functional programmer in me hates stateful objects, so if we want to veer away from them and embrace FP as much as possible, I'm for it too! 😁

Reply:
As long as we're consistent in the usages, I'm fine with either. Thanks!